apm-agent-plugins/apm-profiling-plugin/src/main/java/co/elastic/apm/agent/profiler/asyncprofiler/AsyncProfiler.java

```diff
@@ -102,8 +102,10 @@ private static String getLibraryFileName() {
         String os = System.getProperty("os.name").toLowerCase();
         String arch = System.getProperty("os.arch").toLowerCase();
         if (os.contains("linux")) {
-            if (arch.contains("arm") || arch.contains("aarch")) {
+            if (arch.contains("arm")) {
                 return "libasyncProfiler-linux-arm";
+            } else if (arch.contains("aarch")) {
+                return "libasyncProfiler-linux-aarch64";
```

Comment:
Does this binary work for both 64 and 32-bit aarch? Is there even a 32-bit version? Throw exception if not 64 bit?

Reply:
It's 64-specific. I believe the `os.arch` for `aarch32` should be `arm`. 
@apangin can you confirm?

Reply:
Right, `os.arch` for 32-bit ARM is `arm`, for 64-bit ARM is `aarch64`.
There was even a [fix](http://mail.openjdk.java.net/pipermail/aarch32-port-dev/2016-October/000514.html) to make `os.arch` return `arm` on systems that report `aarch32`.

Reply:
Thanks for confirming! Seems like it should be good enough then, but do you think it's better to look for `aarch64`? Can there be a AArch64 OS where the `os.arch` contains `aarch` but not `aarch64`?

Reply:
IMO, doesn't matter. You may look for `aarch` or `aarch64` - I expect this to be the same set of systems.

Reply:
OK, good enough for me - let's leave that as is. We can change in the future if we get reports this doesn't support values in specific systems.

Reply:
> There was even a fix to make os.arch return arm on systems that report aarch32

Even though this fix is old, I made it more specific so that we do not fail on such old JDKs.